src/components/shared/Entity/DetailsForm/useDataPlaneField.ts

```diff
@@ -25,18 +26,21 @@ interface OneOfElement {
 export default function useDataPlaneField(
     entityType: EntityWithCreateWorkflow
 ) {
-    const dataPlaneIdInURL = useGlobalSearchParams(
-        GlobalSearchParams.DATA_PLANE_ID
-    );
+    const [query, setQuery] = useQueryParams({
+        [GlobalSearchParams.DATA_PLANE_ID]: StringParam,
+    });
+    const dataPlaneIdInURL = query[GlobalSearchParams.DATA_PLANE_ID];
```

Comment:
We can just directly use the query param and read it from there so that we get easy access to the `set` function.